### PR TITLE
Extract RespondWorkflowTaskCompleted handler

### PR DESCRIPTION
**Pull request by stephanos**
## What changed?

Extracted handler for `RespondWorkflowTaskCompleted` out of `workflow_task_handler_callbacks.go`.

## Why?

It's the last one of the WFT handlers that is still in its old place. This completed the refactoring.

## How did you test it?

Pure refactoring, no behavior change.

### CODE REVIEW

File: service/history/api/respondworkflowtaskcompleted/workflow_task_completed_handler.go [hunk 1]
```diff
@@ -65,10 +65,14 @@ import (
...
 
+const (
+	activityCancellationMsgActivityNotStarted = "ACTIVITY_ID_NOT_STARTED"
+)
+
 type (
 	commandAttrValidationFn func() (enumspb.WorkflowTaskFailedCause, error)
 
-	workflowTaskHandlerImpl struct {
+	workflowTaskCompletedHandler struct {
```
Comment (stephanos) [inline service/history/api/respondworkflowtaskcompleted/workflow_task_completed_handler.go:75]: This has been renamed to be more accurate; no other changes were made.

File: service/history/api/respondworkflowtaskcompleted/api_test.go
Comment (stephanos) [inline service/history/api/respondworkflowtaskcompleted/api_test.go:1]: Previously, there were 2 separate test suites for this handler. I have merged them into a single one here.

File: service/history/api/respondworkflowtaskcompleted/api_test.go [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+	"go.temporal.io/server/common/persistence"
+	"go.temporal.io/server/service/history/api"
+	"go.temporal.io/server/service/history/events"
+	"go.temporal.io/server/service/history/hsm"
+	"go.temporal.io/server/service/history/shard"
+	"go.temporal.io/server/service/history/tests"
+	"go.temporal.io/server/service/history/workflow"
+)
+
+type (
```
Comment (stephanos) [inline service/history/api/respondworkflowtaskcompleted/api_test.go:75]: Previously, there were 2 separate test suites for this handler. I have merged them into a single one here.

**Review by alexshtin — APPROVED** [review]

**Review by dandavison — COMMENTED** [review]
Nice!

**Pull request merged**